packages/app-project/src/screens/ProjectHomePage/components/Hero/HeroContainer.js

```diff
@@ -7,84 +7,20 @@ import WideLayout from './components/WideLayout'
 import NarrowLayout from './components/NarrowLayout'
 import fetchWorkflowsHelper from './helpers/fetchWorkflowsHelper'
 
-function storeMapper (stores) {
-  return {
-    activeWorkflows: stores.store.project.links['active_workflows'],
-    defaultWorkflow: stores.store.project.configuration['default_workflow']
-  }
-}
-
 class HeroContainer extends Component {
-  constructor () {
-    super()
-    this.state = {
-      workflows: {
-        loading: asyncStates.initialized,
-        data: []
-      }
-    }
-  }
-
-  componentDidMount () {
-    return this.fetchWorkflows()
-  }
-
-  async fetchWorkflows () {
-    this.setState(state => ({
-      workflows: {
-        ...state.workflows,
-        loading: asyncStates.loading
-      }
-    }))
-    try {
-      const { activeWorkflows, defaultWorkflow, language } = this.props
-      const workflows = await fetchWorkflowsHelper(language, activeWorkflows, defaultWorkflow)
-      this.setState({
-        workflows: {
-          loading: asyncStates.success,
-          data: workflows
-        }
-      })
-    } catch (error) {
-      if (process.browser) {
-        console.error(error)
-      }
-      this.setState(state => ({
-        workflows: {
-          ...state.workflows,
-          loading: asyncStates.error
-        }
-      }))
-    }
-  }
 
   render () {
-    const { workflows } = this.state
-    const { isWide } = this.props
+    console.log('hero props', this.props)
+    const { isWide, workflows } = this.props
+    const workflowData = {
+        loading: asyncStates.success,
```

Comment:
Here I'd hold off on displaying the workflows until login has finished, so this loading state should reflect the user loading state.

Reply:
Resolved by moving loading state down to the workflow selector itself.